### PR TITLE
python310Packages.{fsspec,gcsfs,s3fs}: 2022.8.2 -> 2022.10.0

### PR DESCRIPTION
**Pull request by anthonyroussel**
###### Description of changes

* Upgrade python310Packages.fsspec from 2022.8.2 to 2022.10.0
  * See https://github.com/fsspec/filesystem_spec/blob/master/docs/source/changelog.rst#2022100

* Upgrade python310Packages.gcsfs from 2022.8.2 to 2022.10.0
  * See https://github.com/fsspec/gcsfs/blob/2022.10.0/docs/source/changelog.rst

* Upgrade python310Packages.s3fs from 2022.8.2 to 2022.10.0
  * See https://github.com/fsspec/s3fs/blob/2022.10.0/docs/source/changelog.rst

* Fixed renamed Github repository URL
* Fixed failing auto-upgrade by adding updateScript
  * See https://r.ryantm.com/log/python310Packages.fsspec/2022-11-01.log

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/fsspec/default.nix [hunk 1]
```diff
@@ -13,20 +13,21 @@
 , requests
 , smbprotocol
 , tqdm
+, nix-update-script
 }:
 
 buildPythonPackage rec {
   pname = "fsspec";
-  version = "2022.8.2";
+  version = "2022.10.0";
```
Comment (fabaff) [inline pkgs/development/python-modules/fsspec/default.nix:21]: It's usually required to update [gcsfs](https://github.com/fsspec/gcsfs/blob/2022.10.0/requirements.txt#L6), [s3fs](https://github.com/fsspec/s3fs/blob/2022.10.0/requirements.txt#L2) and perhaps others modules in the process.

File: pkgs/development/python-modules/fsspec/default.nix [hunk 2]
```diff
@@ -66,9 +67,13 @@ buildPythonPackage rec {
     "fsspec"
   ];
 
+  passthru.updateScript = nix-update-script {
+    attrPath = "python3Packages.${pname}";
+  };
+
```
Comment (fabaff) [inline pkgs/development/python-modules/fsspec/default.nix:73]: From my point of view it doesn't make sense to add `nix-update-script` to Python modules. `nix-update` worked fine the past for manual updates, as far as I remember, and it also seems to be picked-up for bulk updates.

File: pkgs/development/python-modules/fsspec/default.nix (hunk 1 above)
Comment (anthonyroussel) [reply pkgs/development/python-modules/fsspec/default.nix:20]: Updated gcsfs and s3fs to 2022.10.0

File: pkgs/development/python-modules/fsspec/default.nix (hunk 2 above)
Comment (anthonyroussel) [reply pkgs/development/python-modules/fsspec/default.nix]: Removed the updateScript passthru.

**Pull request merged**